### PR TITLE
CVSB-9187: logic and unit test implemented

### PR DESCRIPTION
**Pull request by m-ujaffer**
unable to perform integration test because of segment/sub-segment error for AWS xray which is already fixed in CVSB-7964.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (KHodderDVSA) [inline .vscode/launch.json]: Probably shouldn't be pushing these, unless you have a good reason

File: src/services/TestResultsService.ts [hunk 2]
```diff
@@ -283,10 +283,14 @@ export class TestResultsService {
...
-                    if (this.isFirstTestRetestTestType(testType) && dateFns.isEqual(mostRecentExpiryDateOnAllTestTypesByVin, new Date(1970, 1, 1))) {
-                      // Applying CVSB-8658 logic changes if vehicle doesn't currently have an existing expiry date
-                      const regOrFirstUseDate = payload.vehicleType === VEHICLE_TYPES.HGV ? payload.regnDate : payload.firstUseDate;
+                    // Applying CVSB-8658 logic changes if vehicle doesn't currently have an existing expiry date
+                    const regOrFirstUseDate = payload.vehicleType === VEHICLE_TYPES.HGV ? payload.regnDate : payload.firstUseDate;
+                    // preparaing compare date for CVSB-9187 to compare first test/retest conducted after anniversary date
+                    const firstTestAfterAnvCompareDate = dateFns.addYears(dateFns.startOfMonth(regOrFirstUseDate), 1);
+                    // Checks for testType = First test or First test Retest AND if firstUse or registraion date is 1 year from the month of first use.
+                    if (this.isFirstTestRetestTestType(testType) && dateFns.isAfter(new Date(), firstTestAfterAnvCompareDate)) {
```
Comment (agoncearenco) [inline src/services/TestResultsService.ts]: The comment at the line above says that the condition should check if firstUse or registration date is 1 year from the month of first use. But here I see it compared with new Date(). Is that correct?

Comment (m-ujaffer) [reply src/services/TestResultsService.ts]: typo in comment. It should be test date is 1 year from the month of first use or registration date.

File: .vscode/launch.json (hunk 1 above)
Comment (m-ujaffer) [reply .vscode/launch.json]: removed the file. I had updated .gitignore in 7964 but forgot to remove it from this branch.

**Pull request merged**